src/briefcase/images.py

```diff
@@ -0,0 +1,152 @@
+
+    try:
+        # Get the largest available image
+        image = find_largest_png(PosixPath(source).parent)
+
+        #test_split = str(target).split('.')
+        # Determine the image type needed
+        image_type = str(target).split('.')[-1]
+
+        if image_type == 'icns':
+            image = create_icns(target, image)  # Will create temp img artifacts, then write icns file
+
+        else: # Create a single image
+
+            if size is not None:
+                image = image.resize(size)
+
+            # # Make sure the target directory exists
+            target.parent.mkdir(parents=True, exist_ok=True)
+
+            image.save(source)  # Image file saved, so it is now available... or should I use temp here too?
+            source_img_file = source  # Source created, and therefore now available
+
+            # Copy the source image to the target location
+            shutil.copy(source_img_file, target)
+
+    except Exception:
+        import sys
+        the_type, the_value, the_traceback = sys.exc_info()
+        print(f'Exception of type {the_type} occured in Create Image Variant. Value is {the_value}.\n'
+              f'{the_traceback}')
+         #print(
+         #   "Unable to find {source_filename} for {full_role}; using default".format(
+         #       full_role=full_role,  # SHOULD THIS BE PASSED INTO create_image_variant? Is it still relevant?
+         #       source_filename=source,
+         #   )#)
+
+
+def find_largest_png(source: PosixPath) -> Image:
+    """
+    Search a directory of PNGs and returns the widest
+
+    :param source: Directory to search
+    :return: Widest PNG image
+    """
+    largest_image = None
+    img_files = source.glob("*.png")
+    img_files = list(img_files)
+    if img_files:
+
+        try:
+            largest_image = Image.open(img_files.pop())
+
+            for img in img_files:
+
+                maybe_larger_img = Image.open(img)
+
+                # Compare widths
+                if maybe_larger_img.size[0] > largest_image.size[0]:
+                    largest_image = maybe_larger_img
+
+        except OSError:
+            print("Image file cannot be read")
+        except UnidentifiedImageError:
+            print("Image cannot be opened and identified.")
+        except FileNotFoundError:
+            print("Image file cannot be found")
+        except ValueError:
+            print("Mode is not “r”, or a StringIO instance is used for fp")
+        except TypeError:
+            print("If formats is not None, a list or a tuple")
+
+
+    else:
+        print(f"No images in {source}")
+
+    return largest_image
+
+def create_icns(target: PosixPath, image: Image) -> None: #? : # I think it would be much more clear if we could return the .icns file here somehow, then we can only have the final file writing occur in the parent method.
```

Comment:
While a nice idea, it's likely impossible; you're calling out to iconutil, so you don't ever have the icns file in process memory.